packages/svelte-parse/src/main.ts

```diff
@@ -0,0 +1,1219 @@
+				(current_node as Property).value.push(_n as Text);
+				push_node(_n);
+			}
+
+			// capture the token otherwise
+			(current_node as Text).value += value[index];
+
+			chomp();
+			continue;
+		}
+
+		if (current_state === State.IN_DIRECTIVE_SPECIFIER) {
+			if (char === EQUALS) {
+				set_state(State.IN_ATTR_VALUE, true);
+				chomp();
+				continue;
+			}
+
+			if (char === PIPE) {
+				const _n = { value: '', type: 'modifier' };
+				if (generatePositions)
+					//@ts-ignore
+					_n.position = { start: place(), end: {} };
+				(current_node as Directive).modifiers.push(_n as Literal);
+				push_node(_n);
+				set_state(State.IN_ATTR_MODIFIER, true);
+				chomp();
+				continue;
+			}
+
+			let s;
+			// " ", "\n", "/" or ">" => ends the whole thing
+			if (
+				(s = char) === SPACE ||
+				s === TAB ||
+				s === LINEFEED ||
+				s === SLASH ||
+				s === CLOSE_ANGLE_BRACKET
+			) {
+				pop_state();
+				pop_node();
+				continue;
+			}
+
+			(current_node as Directive).specifier += value[index];
+			chomp();
+			continue;
+		}
+
+		if (current_state === State.IN_ATTR_MODIFIER) {
+			if (char === PIPE) {
+				pop_node();
+				const _n = { value: '', type: 'modifier' };
+				if (generatePositions)
+					//@ts-ignore
+					_n.position = { start: place(), end: {} };
+				(current_node as Directive).modifiers.push(_n as Literal);
+				push_node(_n);
+				chomp();
+				continue;
+			}
+
+			if (char === EQUALS) {
+				set_state(State.IN_ATTR_VALUE, true);
+				pop_node();
+				chomp();
+				continue;
+			}
+
+			let s;
+			if ((s = char) === SPACE || s === TAB || s === LINEFEED) {
+				chomp();
+				continue;
+			}
+
+			if ((s = char) === SLASH || s === CLOSE_ANGLE_BRACKET) {
+				pop_node();
+				pop_node();
+				pop_state();
+				continue;
+			}
+			(current_node as Literal).value += value[index];
+			chomp();
+			continue;
+		}
+
+		if (current_state === State.IN_SCRIPT_STYLE) {
+			if (char === OPEN_ANGLE_BRACKET) {
+				if (RE_SCRIPT_STYLE.test(value.substring(index))) {
+					if (generatePositions)
+						//@ts-ignore
+						current_node.position.end = place();
+					pop_node();
+					set_state(State.EXPECT_END_OR_BRANCH, true);
+					continue;
+				}
+			}
+
+			(current_node as SvelteTag).value += value[index];
+			chomp();
+			continue;
+		}
+
+		if (current_state === State.PARSE_CHILDREN) {
+			if (
+				(current_node as SvelteElement | SvelteTag).tagName === 'script' ||
+				(current_node as SvelteElement | SvelteTag).tagName === 'style'
+			) {
+				(current_node as SvelteElement | SvelteTag).type = 'svelteTag';
+				const _n = {
+					type: 'text',
+					value: '',
+				};
+				if (generatePositions)
+					//@ts-ignore
+					_n.position = { start: place(), end: {} };
+				(current_node as SvelteTag).children.push(_n as Text);
+				push_node(_n);
+
+				set_state(State.IN_SCRIPT_STYLE, true);
+				continue;
+			} else {
+				const [children, lastPosition, lastIndex] = childParser({
```

Comment:
minor: destructured assignment can be measurable. if you don't _need_ it I'd do without